pkg/cache/v2/snapshot.go

```diff
@@ -194,3 +209,44 @@ func (s *Snapshot) GetVersion(typeURL string) string {
 	}
 	return s.Resources[typ].Version
 }
+
+// GetVersionMap will construct a verison map of the current snapshot
+func (s *Snapshot) GetVersionMap() (map[string]map[string]string, error) {
+	if s == nil {
+		return nil, fmt.Errorf("missing snapshot")
+	}
+
+	// We need to initialize the map before we process anything since it is nested
+	versionMap := initializeVMap()
+	for i := 0; i < int(types.UnknownType); i++ {
+		typeURL := GetResponseTypeURL(types.ResponseType(i))
+		resources := s.GetResources(typeURL)
```

Comment:
calling GetResourcesAndTtl is a bit cheaper since it avoids rebuilding the return list

Reply:
Well wouldn't we still need to calculate the hash here? Which would require us to end up marshaling the resource anyways so we can rebuild the version map

Reply:
Think I came up with something clever so we don't need to get the resources at all